opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -150,13 +150,18 @@ namespace Opm {
         if (!localWellsActive())
             return;
 
-        if (!param_.matrix_add_well_contributions_) {
-            // if the well contributions are not supposed to be included explicitly in
-            // the matrix, we only apply the vector part of the Schur complement here.
-            for (const auto& well: well_container_) {
-                // r = r - duneC_^T * invDuneD_ * resWell_
-                well->apply(res);
+        if (!param_.matrix_add_well_contributions_)
+        {
+            OPM_BEGIN_PARALLEL_TRY_CATCH();
+            {
+                // if the well contributions are not supposed to be included explicitly in
+                // the matrix, we only apply the vector part of the Schur complement here.
+                for (const auto& well: well_container_) {
+                    // r = r - duneC_^T * invDuneD_ * resWell_
+                    well->apply(res);
+                }
             }
+            OPM_END_PARALLEL_TRY_CATCH("BlackoilWellModel::linearize failed: ")
             return;
```

Comment:
This TRY_CATCH creates problems when combined with https://github.com/OPM/opm-simulators/pull/3571
After fixing the rebase by added the (... , grid().comm()) Any model abort when run in parallel. Removing this TRY/CATCH is sufficient to fix the "issue".

Reply:
my fault should have made sure that all processes reach OPM_END_PARALLEL_TRY_CATCH. Not the case if some have no wells.

Reply:
Please cherry-pick fc7609bfd2 and this should be fixed. OPM#3571 is unrelated to this, my dumbness is not. :wink: